### PR TITLE
Search term in url now populates search input

### LINKED ISSUE #586: A libraries search by q query-string parameter does not show search text
## Problem
Cljdoc supports searching via a `q` query-string parameter.

## Context
This path can be invoked by anyone, but is specifically invoked when a user is using OpenSearch from their browser and selects a suggestion (see #581 for an overview of how OpenSearch works from a web browser).

## Actual Behavior
The query text does not show up in the search box:

## Expect Behavior
I would expect to see `rewrite` in the libraries search text box.

### PR DESCRIPTION
**Pull request by lread**
Got some inspiration from single docset search.

Not sure why we don't grab query string q param directly from url in JS:
Server side render detects q param and populates data-input-value
attribute on search input elem. Maybe there's a good reason, did not change.

Also: when search input is cleared by user, now closing results.

Closes #586

### CODE REVIEW

File: js/search.tsx [hunk 1]
```diff
@@ -79,11 +80,14 @@ const refineSearchResults = (raw: RawSearchResults): SearchResults => ({
...
-    .then(response => response.json())
-    .then((json: RawSearchResults) => cb(refineSearchResults(json).results));
+  if (!str) {
+    cb([]);
+  } else {
+    const uri = "/api/search?q=" + str; //+ "&format=json";
+    fetch(uri)
+      .then(response => response.json())
+      .then((json: RawSearchResults) => cb(refineSearchResults(json).results));
+  }
```
[the remark this answers is not in the document]
Comment (corasaurus-hex) [reply js/search.tsx:90]: probably even better is to do:

```javascript
const uri = new URL(window.location.origin);
uri.pathname = "/api/search";
uri.searchParams.set('q', str);
fetch(uri)
```

[the remark this answers is not in the document]
Comment (corasaurus-hex) [reply js/search.tsx:90]: I know this wasn't added just now, it just stuck out to me when I saw this PR.

[the remark this answers is not in the document]
Comment (lread) [reply js/search.tsx:90]: Thanks for noticing this old code @corasaurus-hex!
Please feel free to make any changes that make sense to you.

[the remark this answers is not in the document]
Comment (lread) [reply js/search.tsx:90]: Every so often I get the itch, as we chatted about a while ago, to switch to from ts to cljs. But I haven't scratched it yet!

**Pull request merged**